crates/nu-engine/src/closure_eval.rs

```diff
@@ -77,6 +88,14 @@ impl ClosureEval {
         }
     }
 
+    /// Sets whether to enable debugging when evaluating the closure.
+    ///
+    /// By default, this is controlled by the [`EngineState`] used to create this [`ClosureEval`].
+    pub fn debug(&mut self, debug: bool) -> &mut Self {
+        self.eval = eval_fn(debug);
+        self
+    }
+
```

Comment:
@kubouch Should `engine_state.is_debugging()` always be used, or are there situations where the debugging status needs to be overwritten? E.g., `env.rs` and `style_computer.rs` used `eval_block::<WithoutDebug>` instead of the `get_eval_block` function (see [48d0e46](https://github.com/nushell/nushell/pull/12541/commits/48d0e46d8b20c9dbdf7a5044a26781abfb652daf)). Basically, should I keep this function?

Reply:
Debugging status is _only_ set in commands that run the debugger, which currently is _only_ the `profile` command. I don't have time to check the rest of this PR, but I'll try later.

Reply:
I checked and your usage of this method seems fine :+1: . By default, it inherits the debugging information from the EngineState and you explicitly turn it off where we previously called `eval_block::<WithoutDebug>()`.